pkg/reconciler/v1alpha1/taskrun/taskrun.go

```diff
@@ -270,3 +276,23 @@ func applyParameters(b *buildv1alpha1.Build, t *v1alpha1.Task, tr *v1alpha1.Task
 
 	return builder.ApplyReplacements(b, replacements)
 }
+
+func applyResources(b *buildv1alpha1.Build, t *v1alpha1.Task, tr *v1alpha1.TaskRun, lister listers.PipelineResourceLister) (*buildv1alpha1.Build, error) {
```

Comment:
1. ive been finding its easier to write tests for tasks that take `Listers` if you take functions that return objects instead, e.g.:
  https://github.com/knative/build-pipeline/blob/167136a3aae64835ecb526d8426180f350367b5b/pkg/reconciler/v1alpha1/pipelinerun/resources/pipelinestate.go#L71-L82

2. What do you think about moving this function out of `taskrun.go` into maybe the `resources` package?

Reply:
i think having tests specific to this function will be easier especially as we add more resources

Reply:
Done!